src/datepicker/datepicker.component.ts

```diff
@@ -199,7 +202,9 @@ export class DatePicker implements OnDestroy, OnChanges, AfterViewChecked {
 
 	protected flatpickrInstance = null;
 
-	constructor(protected elementRef: ElementRef) { }
+	protected visibilitySubscription = new Subscription();
+
+	constructor(protected elementRef: ElementRef, protected elementService: ElementService = null) { }
```

Comment:
```suggestion
	constructor(protected elementRef: ElementRef, protected elementService: ElementService) { }
```

You shouldn't need to null out the service 🤔